devtools/x/src/build.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    cargo::{CargoArgs, CargoCommand},
+    context::XContext,
+    utils,
+    utils::project_root,
+    Result,
+};
+use log::info;
+use std::{
+    ffi::OsString,
+    process::{Command, Stdio},
+};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct Args {
+    #[structopt(long, short, number_of_values = 1)]
+    /// Run test on the provided packages
+    package: Vec<String>,
+    #[structopt(long, short)]
+    /// Skip running expensive libra testsuite integration tests
+    unit: bool,
+    #[structopt(long)]
+    /// Test only this package's library unit tests, skipping doctests
+    lib: bool,
+    #[structopt(long)]
+    /// Do not fast fail the run if tests (or test executables) fail
+    no_fail_fast: bool,
+    #[structopt(long)]
+    /// Do not run tests, only compile the test executables
+    no_run: bool,
+    #[structopt(long, short)]
+    /// Number of parallel jobs, defaults to # of CPUs
+    jobs: Option<u16>,
+    #[structopt(long, parse(from_os_str))]
+    /// Directory to output HTML coverage report to (using grcov)
+    html_cov_dir: Option<OsString>,
+    #[structopt(name = "TESTNAME", parse(from_os_str))]
+    testname: Option<OsString>,
+    #[structopt(name = "ARGS", parse(from_os_str), last = true)]
+    args: Vec<OsString>,
```

Comment:
This looks like it was mostly copied from `xtest` and there's a couple of these params that maybe don't make sense wrt wrapping `cargo build`.

Reply:
ah didn't mean to include this yet.